### DISCUSSION TITLE
OpenTelemetry tracing.

### DISCUSSION THREAD

**Discussion by xavierjohn** [vkhorikov/CSharpFunctionalExtensions#538] [General]
I've been considering which tracing should be enabled from a production standpoint. From a telemetry perspective, should we classify moving to the failure track as a failure? 
In the example below, the trace shows that 'Ensure' has moved to the failure track, but this may be part of normal business operations. Should we still mark it as a failure?
From looking at the production log, I can see were things have moved to the failed track. Some people may consider this as just noise.

![image](https://github.com/vkhorikov/CSharpFunctionalExtensions/assets/1859710/0604eaf6-d272-461b-beac-748a09ac6f8a)

**Comment by vkhorikov** [comment]
Unless the goal is to measure performance, this type of telemetry is too low-level and fine-grained. I'd recommend stepping a level higher than that.

**Reply by xavierjohn** [reply]
True, this will mostly not be needed. Sometimes people log every step and for such cases, they can temporarily turn on the library as a source. It will tell which ROP function failed instead of having to add explicit logging.
